### ISSUE TITLE
Create High Level Test Framework for StandardRuleset.ResolveMoves(...)

### ISSUE THREAD

**Issue by bvanvugt** [BattlesnakeOfficial/rules#3]
We're hoping to build a very thorough and thoughtful test suite for high level move resolution (using the Standard Ruleset for now).

Ideally this consumes input board state, input moves, and expected output board state from test files (probably json?).

Example:
tests/standard/1.in.json 
---> contains a JSON representation of BoardState to start with + SnakeMoves to execute
tests/standard/1.out.json
---> contains a JSON representation of the resulting BoardState after StandardRuleset is applied

**Comment by xtagon** [comment]
I'd love to collaborate on this. Ideally the test data/fixtures could be made public so that the same tests can run for our snakes as the tests for the official rules.

What's the next step and how can I help?

**Comment by bvanvugt** [comment]
Yes - exactly! 

Next step I think would be a single test case, maybe in standard_test.go, that loads a board state from a json file and checks the results against a second json file. Once we have one test working, we can worry about adding others.

**Comment by xtagon** [comment]
Interesting, so in theory when it's time to build the whole canonical dataset we would run the rulesets for lots of scenarios, save the results as JSON, and then store it for history? So after that exists, it's literally just a content comparison to test it. That means testing it wouldn't even have to load up the full example JSON, it could go by a hash or file checksum

**Comment by bvanvugt** [comment]
Yeah, regression testing becomes super easy. New rulesets can be implemented using test cases first. In theory yeah you could use a checksum but ordering in JSON isn't deterministic so that would need additional manipulation. Easier to just do JSON object dumps of the structs, load and run compares (which go handles well).

**Comment by bvanvugt** [comment]
(closed by accident ;))

**Comment by xtagon** [comment]
Haven't really started yet, but saving this here for reference: https://www.npmjs.com/package/json-diff (might be useful)
